n2survey/lime/survey.py

```diff
@@ -531,6 +609,104 @@ def plot(
 
         return fig, ax
 
+    def check_plot_implemented(self, question, compare_with=None, add_questions=None):
+        """
+        function to check if question type and/or combination of questions for
+        compare_with is already implemented and working
+        """
+        if not add_questions:
+            add_questions = []
+        supported_plots = ["single-choice", "multiple-choice", "array"]
+        supported_comparisons = [("single-choice", "single-choice")]
+        all_plots = [question]
+        if compare_with:
+            all_plots.append(compare_with)
+        for i in add_questions:
+            all_plots.append(i)
+        for plot in all_plots:
+            if self.get_question_type(plot) not in supported_plots:
```

Comment:
all_plots-supported_plots gives back:
unsupported operand type(s) for -: 'list' and 'list'

Reply:
You get "unsupported operand types" because one needs to convert both into sets, with `set(all_plots)` for instance. Sets are unordered collection of unique items, as in maths. If the list has repeated items, only one would remain after conversion, and then one can perform set operations with `+` and `-` operators to find union, intersection, difference etc.